llvm/test/CodeGen/AIE/aie2/GlobalISel/prelegalizercombiner-shufflevector-buildvector.mir

```diff
@@ -0,0 +1,322 @@
+    ; CHECK-NEXT: [[SHUF:%[0-9]+]]:_(<16 x s32>) = G_SHUFFLE_VECTOR [[COPY]](<8 x s32>), [[DEF]], shufflemask(12, 13, 11, 15, 9, 9, 8, 12, 8, 8, 13, 12, 12, 10, 8, 9)
+    ; CHECK-NEXT: $x0 = COPY [[SHUF]](<16 x s32>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $x0
+    %0:_(s32) = G_CONSTANT i32 28
+    %1:_(s32) = G_CONSTANT i32 24
+    %2:_(s32) = G_CONSTANT i32 18
+    %3:_(s32) = G_CONSTANT i32 8
+    %4:_(s32) = G_CONSTANT i32 7
+    %5:_(s32) = G_CONSTANT i32 3
+    %6:_(s32) = G_CONSTANT i32 2
+    %7:_(s32) = G_CONSTANT i32 1
+    %8:_(<8 x s32>) = COPY $wl2
+    %9:_(<8 x s32>) = G_IMPLICIT_DEF
+    %10:_(<16 x s32>) = G_SHUFFLE_VECTOR %8:_(<8 x s32>), %9:_, shufflemask(12, 13, 11, 15, 9, 9, 8, 12, 8, 8, 13, 12, 12, 10, 8, 9)
+    $x0 = COPY %10:_(<16 x s32>)
+    PseudoRET implicit $lr, implicit $x0
+...
+---
+name:            shufflevector_undef_reg
+legalized:       false
+body:             |
+  bb.1.entry:
+    liveins: $wl2, $wl4
+    ; CHECK-LABEL: name: shufflevector_undef_reg
+    ; CHECK: liveins: $wl2, $wl4
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[DEF:%[0-9]+]]:_(<8 x s32>) = G_IMPLICIT_DEF
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(<8 x s32>) = COPY $wl2
+    ; CHECK-NEXT: [[SHUF:%[0-9]+]]:_(<16 x s32>) = G_SHUFFLE_VECTOR [[DEF]](<8 x s32>), [[COPY]], shufflemask(12, 13, 11, 15, 9, 9, 8, 12, 8, 8, 13, 12, 12, 10, 8, 9)
+    ; CHECK-NEXT: $x0 = COPY [[SHUF]](<16 x s32>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $x0
+    %0:_(s32) = G_CONSTANT i32 28
+    %1:_(s32) = G_CONSTANT i32 24
+    %2:_(s32) = G_CONSTANT i32 18
+    %3:_(s32) = G_CONSTANT i32 8
+    %4:_(s32) = G_CONSTANT i32 7
+    %5:_(s32) = G_CONSTANT i32 3
+    %6:_(s32) = G_CONSTANT i32 2
+    %7:_(s32) = G_CONSTANT i32 1
+    %8:_(<8 x s32>) = G_IMPLICIT_DEF
+    %9:_(<8 x s32>) = COPY $wl2
+    %10:_(<16 x s32>) = G_SHUFFLE_VECTOR %8:_(<8 x s32>), %9:_, shufflemask(12, 13, 11, 15, 9, 9, 8, 12, 8, 8, 13, 12, 12, 10, 8, 9)
+    $x0 = COPY %10:_(<16 x s32>)
+    PseudoRET implicit $lr, implicit $x0
+...
+---
+name:            shufflevector_broadcast
+legalized:       false
+body:             |
+  bb.1.entry:
+    liveins: $r0
+    ; CHECK-LABEL: name: shufflevector_broadcast
+    ; CHECK: liveins: $r0
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(s32) = COPY $r0
+    ; CHECK-NEXT: [[BUILD_VECTOR:%[0-9]+]]:_(<8 x s32>) = G_BUILD_VECTOR [[COPY]](s32), [[COPY]](s32), [[COPY]](s32), [[COPY]](s32), [[COPY]](s32), [[COPY]](s32), [[COPY]](s32), [[COPY]](s32)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit [[BUILD_VECTOR]](<8 x s32>)
+    %1:_(s32) = COPY $r0
+    %3:_(<8 x s32>) = G_IMPLICIT_DEF
+    %5:_(s32) = G_IMPLICIT_DEF
+    %2:_(<8 x s32>) = G_BUILD_VECTOR %1(s32), %5(s32), %5(s32), %5(s32), %5(s32), %5(s32), %5(s32), %5(s32)
+    %0:_(<8 x s32>) = G_SHUFFLE_VECTOR %2(<8 x s32>), %3, shufflemask(0, 0, 0, 0, 0, 0, 0, 0)
+    PseudoRET implicit $lr, implicit %0
```

Comment:
Looks like this is missing the closing `...` at the end, and then I think this would fail, because this is further combined into `G_AIE_BROADCAST`?

Reply:
The shufflevector branch has become really divergent in the past month or so, so that hasn't landed yet. Let's try to get all of that merged as quickly as we can, I'll manually merge them to make sure that spectests are updated before enter the main branch.

Missing the closing `...` is deliberate, trailing ellipses have caused errors with the unit tests scripts for me before.

Reply:
Ran into it again just now. If you have a trailing ellipsis, you need a trailing newline, or otherwise the update mir script will error out. If you skip it, then it doesn't matter whether you have a newline or not. That is why I tend to commit mine without the trailing ellipsis.